components/NavigationDrawer/tests/unit/MDCNavigationDrawerScrollViewTests.m

```diff
@@ -613,30 +612,65 @@ - (void)testAddedHeightWithTrackingScrollView {
   XCTAssertEqualWithAccuracy(newContentViewControllerHeight, contentViewControllerHeight, 0.001);
 }
 
-- (void)testCalculateInitialDrawerFactorWithSmallHeight {
+- (void)testCalculateInitialDrawerHeightWithSmallHeight {
   // Given
   CGRect fakeRect = CGRectMake(0, 0, 250, 500);
   self.fakeBottomDrawer.originalPresentingViewController.view.bounds = fakeRect;
   self.fakeBottomDrawer.contentViewController.preferredContentSize = CGSizeMake(250, 100);
 
   // When
-  CGFloat drawerFactor = [self.fakeBottomDrawer calculateInitialDrawerFactor];
+  CGFloat drawerHeight = [self.fakeBottomDrawer calculateMaximumInitialDrawerHeight];
 
   // Then
-  XCTAssertEqualWithAccuracy(drawerFactor, 0.2, 0.001);
+  XCTAssertEqualWithAccuracy(drawerHeight, 100, 0.001);
 }
 
-- (void)testCalculateInitialDrawerFactorWithLargeHeight {
+- (void)testCalculateInitialDrawerHeightWithLargeHeight {
   // Given
   CGRect fakeRect = CGRectMake(0, 0, 250, 500);
   self.fakeBottomDrawer.originalPresentingViewController.view.bounds = fakeRect;
   self.fakeBottomDrawer.contentViewController.preferredContentSize = CGSizeMake(250, 1000);
 
   // When
-  CGFloat drawerFactor = [self.fakeBottomDrawer calculateInitialDrawerFactor];
+  CGFloat drawerHeight = [self.fakeBottomDrawer calculateMaximumInitialDrawerHeight];
 
   // Then
-  XCTAssertEqualWithAccuracy(drawerFactor, 0.5, 0.001);
+  XCTAssertEqualWithAccuracy(drawerHeight, 250, 0.001);
+}
+
+- (void)testSettingMaximumInitialDrawerHeight {
+  // Given
+  self.drawerViewController.maximumInitialDrawerHeight = 500;
+
+  // Then
+  MDCBottomDrawerPresentationController *presentationController =
+      (MDCBottomDrawerPresentationController *)self.drawerViewController.presentationController;
+  XCTAssertEqualWithAccuracy(500, presentationController.maximumInitialDrawerHeight, 0.001);
+}
+
+- (void)testInitialDrawerHeight {
+  // Given
+  CGRect fakeRect = CGRectMake(0, 0, 250, 500);
+  self.fakeBottomDrawer.maximumInitialDrawerHeight = 320;
+  self.fakeBottomDrawer.originalPresentingViewController.view.bounds = fakeRect;
+  self.fakeBottomDrawer.contentViewController.preferredContentSize = CGSizeMake(250, 1000);
+
+  // When
+  CGFloat drawerHeight = [self.fakeBottomDrawer calculateMaximumInitialDrawerHeight];
+
+  // Then
+  XCTAssertEqualWithAccuracy(drawerHeight, 320, 0.001);
+}
```

Comment:
Please also add a test validating the inversion here of `maximumInitialDrawerHeight = 1000` and a `preferredContentSize.height of 320`, with the expected outcome being 320.